### PR TITLE
adding annotation information to set sidecar container's seccompprofiletype

### PR DESCRIPTION
**Pull request by JaydipGabani**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description
Adding information about annotation `dapr.io/sidecar-seccomp-profile-type` introduced with [this PR](https://github.com/dapr/dapr/pull/5945)

## Issue reference
#[dapr/5916](https://github.com/dapr/dapr/issues/5916)

### CODE REVIEW

**Comment by ItalyPaleAle** [comment]
LGTM. Can you also please add a sentence in the Kubernetes production page? https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-production/#recommended-security-configuration

**Comment by JaydipGabani** [comment]
> LGTM. Can you also please add a sentence in the Kubernetes production page? https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-production/#recommended-security-configuration

Could you provide a specific location as to where to add the sentence and what the sentence should say? you linked `recommended-security-configuration` section of the page which has nothing related to sidecar container configurations in it. I see https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-production/#sidecar-resource-settings, but that section is more related to how to define how much resources can sidecar consume, also not related to sidecar's securityContext. so I am a little confused on where and what to add on Kubernetes production page

File: daprdocs/content/en/reference/arguments-annotations-overview.md [hunk 1]
```diff
@@ -66,3 +66,4 @@ This table is meant to help users understand the equivalent options for running
 | not supported | not supported | | `dapr.io/volume-mounts` | List of pod volumes to be mounted to the sidecar container in read-only mode. Strings consisting of `volume:path` pairs separated by a comma. Example, `"volume-1:/tmp/mount1,volume-2:/home/root/mount2"`. |
 | not supported | not supported | | `dapr.io/volume-mounts-rw` | List of pod volumes to be mounted to the sidecar container in read-write mode. Strings consisting of `volume:path` pairs separated by a comma. Example, `"volume-1:/tmp/mount1,volume-2:/home/root/mount2"`. |
 | `--disable-builtin-k8s-secret-store` | not supported | | `dapr.io/disable-builtin-k8s-secret-store` | Disables BuiltIn Kubernetes secret store. Default value is false. See [Kubernetes secret store component]({{<ref "kubernetes-secret-store.md">}}) for details. |
+| not supported | not supported | | `dapr.io/sidecar-seccomp-profile-type` | Set sidecar container's `securityContext.seccompProfile.type` to `Unconfined`, `RuntimeDefault`, or `Localhost`. By default, the field will be ommited from sidecar container. |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/arguments-annotations-overview.md]:
```suggestion
| not supported | not supported | | `dapr.io/sidecar-seccomp-profile-type` | Set sidecar container's `securityContext.seccompProfile.type` to `Unconfined`, `RuntimeDefault`, or `Localhost`. By default, the field will be omitted from sidecar container. |
```

**Comment by hhunter-ms** [comment]
> > LGTM. Can you also please add a sentence in the Kubernetes production page? https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-production/#recommended-security-configuration
> 
> Could you provide a specific location as to where to add the sentence and what the sentence should say? you linked `recommended-security-configuration` section of the page which has nothing related to sidecar container configurations in it. I see https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-production/#sidecar-resource-settings, but that section is more related to how to define how much resources can sidecar consume, also not related to sidecar's securityContext. so I am a little confused on where and what to add on Kubernetes production page

@ItalyPaleAle what are your thoughts on this?

**Comment by ItalyPaleAle** [comment]
@JaydipGabani I was thinking a section to cover the Seccomp Profile as well as the new support for dropping capabilities. Similar to this: https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-production/#service-account-tokens WDYT?

File: daprdocs/content/en/reference/arguments-annotations-overview.md (hunk 1 above)
Comment (msfussell) [inline daprdocs/content/en/reference/arguments-annotations-overview.md]: For this sentence, 

"By default, the field will be omitted from sidecar container."
Don't you mean to say
"By default, this annotation is not set on the Dapr sidecar."

**Comment by msfussell** [comment]
@JaydipGabani - What @ItalyPaleAle is referring to, is that there needs to be a section in the docs that describe the "why" and "when" to use this annotation. Having this in the annotation file is not enough since it does not describe a use case and best practices for when to set this. If you read through the whole issue on security settings that brought you to implement this to start with, how do we capture that explanation in the docs for developers.

**Comment by JaydipGabani** [comment]
@ItalyPaleAle @msfussell Thanks for making things clear. I am working on changes will update PR by tomorrow

**Comment by JaydipGabani** [comment]
@ItalyPaleAle @msfussell PTAL, I made the appropriate changes

**Comment by ItalyPaleAle** [comment]
Please keep this open!

**Comment by JaydipGabani** [comment]
@ItalyPaleAle @yaron2 wondering whats the reason for keeping this PR open?

**Comment by ItalyPaleAle** [comment]
@JaydipGabani a @dapr/maintainers-docs will need to review this, I am not a maintainer here :(

**Review by hhunter-ms — APPROVED** [review]
lgtm

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md [hunk 2]
```diff
@@ -236,6 +236,12 @@ The injector watchdog is disabled by default when running Dapr in Kubernetes mod
 
 Refer to the documentation for the [Dapr operator]({{< ref operator >}}) service for more details on the injector watchdog and how to enable it.
 
+## Configuring seccompProfile for sidecar containers
+
+By default, Dapr sidecar injector injects a sidecar without any `seccompProfile`. However, to have Dapr sidecar contianer run successfully in a namespace with [Restricted](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted) profile, sidecar container needs to have `securityContext.seccompProfile.Type` to not be `nil`.
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md]:
```suggestion
By default, the Dapr sidecar Injector injects a sidecar without any `seccompProfile`. However, to have Dapr sidecar container run successfully in a namespace with [Restricted](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted) profile, the sidecar container needs to have `securityContext.seccompProfile.Type` to not be `nil`. 
```

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md [hunk 3]
```diff
@@ -236,6 +236,12 @@ The injector watchdog is disabled by default when running Dapr in Kubernetes mod
 
 Refer to the documentation for the [Dapr operator]({{< ref operator >}}) service for more details on the injector watchdog and how to enable it.
 
+## Configuring seccompProfile for sidecar containers
+
+By default, Dapr sidecar injector injects a sidecar without any `seccompProfile`. However, to have Dapr sidecar contianer run successfully in a namespace with [Restricted](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted) profile, sidecar container needs to have `securityContext.seccompProfile.Type` to not be `nil`. 
+
+Refer to [this]({{< ref "arguments-annotations-overview.md" >}}) documentation to set appropriate `seccompProfile` on sidecar container accordingly to which profile it would be running with.
```
Comment (msfussell) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md]:
```suggestion
Refer to [this]({{< ref "arguments-annotations-overview.md" >}}) documentation to set appropriate `seccompProfile` on sidecar container according to which profile it is running with.
```

File: daprdocs/content/en/reference/arguments-annotations-overview.md [hunk 4]
```diff
@@ -66,3 +66,4 @@ This table is meant to help users understand the equivalent options for running
 | not supported | not supported | | `dapr.io/volume-mounts` | List of pod volumes to be mounted to the sidecar container in read-only mode. Strings consisting of `volume:path` pairs separated by a comma. Example, `"volume-1:/tmp/mount1,volume-2:/home/root/mount2"`. |
 | not supported | not supported | | `dapr.io/volume-mounts-rw` | List of pod volumes to be mounted to the sidecar container in read-write mode. Strings consisting of `volume:path` pairs separated by a comma. Example, `"volume-1:/tmp/mount1,volume-2:/home/root/mount2"`. |
 | `--disable-builtin-k8s-secret-store` | not supported | | `dapr.io/disable-builtin-k8s-secret-store` | Disables BuiltIn Kubernetes secret store. Default value is false. See [Kubernetes secret store component]({{<ref "kubernetes-secret-store.md">}}) for details. |
+| not supported | not supported | | `dapr.io/sidecar-seccomp-profile-type` | Set sidecar container's `securityContext.seccompProfile.type` to `Unconfined`, `RuntimeDefault`, or `Localhost`. By default, this annotation is not set on the Dapr sidecar, hence the filed will be omitted from sidecar container. |
```
Comment (msfussell) [inline daprdocs/content/en/reference/arguments-annotations-overview.md]:
```suggestion
| not supported | not supported | | `dapr.io/sidecar-seccomp-profile-type` | Set the sidecar container's `securityContext.seccompProfile.type` to `Unconfined`, `RuntimeDefault`, or `Localhost`. By default, this annotation is not set on the Dapr sidecar, hence the field is omitted from sidecar container. |
```

**Review by msfussell — APPROVED** [review]
Did some minor spelling and grammar updates

**Pull request merged**